x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -426,23 +426,43 @@ public void writeTo(StreamOutput out) throws IOException {
 
     public static RoleDescriptor parse(String name, BytesReference source, boolean allow2xFormat, XContentType xContentType)
         throws IOException {
+        return parse(name, source, allow2xFormat, xContentType, true);
+    }
+
+    public static RoleDescriptor parse(
+        String name,
+        BytesReference source,
+        boolean allow2xFormat,
+        XContentType xContentType,
+        boolean allowWorkflowsRestriction
```

Comment:
Nit: Can we rename this to just `allowRestriction` since it is actually about the top level `restriction` field? The same nit goes for other places where this variable is defined.